src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseServiceIntegrationTest.java

```diff
@@ -185,6 +189,24 @@ private void testSearchTermMatchesId() throws Exception {
         assertThat(searchResult.getResultsOnPage()).hasSize(1);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    void testCourseAndExamFiltersAsInstructor() throws Exception {
+        database.addCourseWithNamedProgrammingExerciseAndTestCases("Ankh");
+        database.addCourseExamExerciseGroupWithOneProgrammingExercise("Ankh-Morpork", "AnkhMorpork");
+
+        exerciseIntegrationTestUtils.testCourseAndExamFilters("/api/programming-exercises/");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void testCourseAndExamFiltersAsAdmin() throws Exception {
+        database.addCourseWithNamedProgrammingExerciseAndTestCases("Ankh");
+        database.addCourseExamExerciseGroupWithOneProgrammingExercise("Ankh-Morpork", "AnkhMorpork");
+
+        exerciseIntegrationTestUtils.testCourseAndExamFilters("/api/programming-exercises/");
+    }
+
```

Comment:
Code duplication

Reply:
> I don't like that you copied the code five times. Can this be avoided?

I see no possibility without the scope of this PR exploding.

Reply:
@canberkanar I disagree in this case. Test with separate roles -> Separate Tests. Tests should only test one specific scenario